.gitlab/builds.gitlab-ci.yml

```diff
@@ -12,8 +12,9 @@
   script:
     - git clone https://github.com/flux-framework/flux-core
     - cd flux-core
-    - export FLUX_BUILD_DIR=$(pwd)
     - ./autogen.sh
-    - ./configure
-    - make -j 32
-    - cd ..
+    - mkdir build
+    - cd build
+    - ../configure --prefix=$(pwd)
```

Comment:
I would not have the install prefix and build dir be the same directory, as it could get confusing. You could install into an `inst` or `install` subdirectory though...

Reply:
Apologies, I'm still a bit confused on the difference in `make` and `make install`. Are you proposing something like:
```
mkdir build
cd build
mkdir install
../configure --prefix=$(pwd)/install
make -j $(nproc) install
```

Reply:
And then (if I'm understanding this right) the binaries such as `flux` will be in `build/src/cmd`, while the pkgconfig files will be in `build/install/lib/pkgconfig`?

Reply:
> Apologies, I'm still a bit confused on the difference in make and make install

`make` just compiles a project, building all executables and libraries etc, but leaving them in place in the build directory. `make install` installs a project into the system or prefix you configured it for (think of it like installing a package).

More info here: https://www.gnu.org/software/autoconf/manual/autoconf-2.63/html_node/Basic-Installation.html

Reply:
> And then (if I'm understanding this right) the binaries such as flux will be in build/src/cmd

No, `make install` will _install_ the project to the specified `--prefix` using a set of standard paths that linux distros expect, unless those paths were also overridden on the command line. So `flux` will be in `$prefix/usr/bin/flux` (`src/cmd/flux` is the path to the built flux before installation).

Maybe an intro to autotools and the Linux filesystem hierarchy standard would be useful reading, e.g.

https://thoughtbot.com/blog/the-magic-behind-configure-make-make-install